terraform/provider.tf

```diff
@@ -2,12 +2,11 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 3.74"
+      version = "~> 4.1"
     }
   }
 }
 
 provider "aws" {
-  profile = "default"
```

Comment:
It seems like support for this was removed in v4, and it’s now looking for an actual profile named `default`. Removing the property altogether seems to have fixed the issue. (The best pointer/hint I could find on this was here: https://github.com/GoogleCloudPlatform/terraformer/issues/940)